### PR TITLE
helm: Add envoyConfig value

### PR DESCRIPTION
**Pull request by jrajahalme**
Add envoyConfig value to be able to enable CiliumEnvoyConfig with access to a secrets namespace in cases where Cilium Ingress nor GatewayAPI is enabled, or a different secrets namespace is needed.

envoyConfig.secretsNamespace.name defines the namespace to which Cilium Agent is given read access to.

```release-note
Add helm value `envoyConfig.enabled` that can be used to enable CiliumEnvoyConfig CRD independently of Cilium Ingress controller.
```

### CODE REVIEW

File: install/kubernetes/cilium/templates/cilium-secrets-namespace.yaml
Comment (youngnick) [inline install/kubernetes/cilium/templates/cilium-secrets-namespace.yaml:26]: I feel like I ran into a complexity limit in my brain evaluating this.

**Review by youngnick — APPROVED** [review]
LGTM, although I wish there was a way to make that nested logic in the big IF easier to read.

File: install/kubernetes/cilium/templates/cilium-secrets-namespace.yaml
Comment (thorn3r) [reply install/kubernetes/cilium/templates/cilium-secrets-namespace.yaml:26]: like it or not, this is what peak helm performance looks like :runner:

**Review by thorn3r — APPROVED** [review]
had to up my caffeine intake to get through that helm logic, but looks solid :+1:

File: install/kubernetes/cilium/templates/cilium-secrets-namespace.yaml
Comment (kaworu) [reply install/kubernetes/cilium/templates/cilium-secrets-namespace.yaml:26]: is-this-lisp

On a more serious note maybe defining some variables could help readability e.g. 

```
{{- $ingressControllerCreateSecret := (and .Values.ingressController.enabled .Values.ingressController.secretsNamespace.create .Values.ingressController.secretsNamespace.name) -}}
```

File: install/kubernetes/cilium/templates/cilium-secrets-namespace.yaml
Comment (kaworu) [inline install/kubernetes/cilium/templates/cilium-secrets-namespace.yaml:22]: I think we can remove this `and` nesting as we're already at the top-level `and` here correct?

File: install/kubernetes/cilium/templates/cilium-secrets-namespace.yaml
Comment (squeed) [reply install/kubernetes/cilium/templates/cilium-secrets-namespace.yaml:26]: there's a better way to do this :-)

Make a dict of desired secret namespace names, something like

```

{{- $secretNamespaces := dict -}}
{{- range $cfg := tuple .Values.ingressController .Values.gatewayAPI .Values.envoyConfig -}}
{{- if and $cfg.enabled $cfg.secretsNamespace.create $cfg.secretsNamespace.name -}}
{{- $_ = set $secretNamespaces $cfg.secretsNamespace.name -}}
{{- end -}}
{{- end -}}

{{- range $name, _ := $secretNamespaces }}
---
apiVersion: v1
kind: Namespace
metadata:
  name: {{ $name | quote }}
{{- end}}
```

File: install/kubernetes/cilium/values.yaml.tmpl
Comment (squeed) [inline install/kubernetes/cilium/values.yaml.tmpl:656]: Bringing feedback from the other PR, the conventional way of indicating that a value has a runtime-computed default is to use nil / `~`.

This also allows the end-user to force-override the defaulting logic in either direction.

**Review by kaworu — APPROVED** [review]
Couple of nitpicking comments, other than that LGTM.

File: install/kubernetes/cilium/values.yaml.tmpl
Comment (zacharysarah) [inline install/kubernetes/cilium/values.yaml.tmpl:663]: Nits for clarity
```suggestion
    # -- The name of the secret namespace to which Cilium agents are given read access.
```

File: install/kubernetes/cilium/values.yaml.tmpl
Comment (zacharysarah) [inline install/kubernetes/cilium/values.yaml.tmpl:1678]: No Latin abbreviations
```suggestion
    #   - disabled: Disable L7 load balancing by way of service annotation.
```

**Review by zacharysarah — APPROVED** [review]
@jrajahalme 👋🏻 Some clarity nits, otherwise LGTM. Approving with the understanding that changes are required prior to merge.

**Comment by michi-covalent** [comment]
@jrajahalme do you think this is a blocker for June patch releases, or can it wait until July? we usually target 15th as the release date.

**Comment by jrajahalme** [comment]
Rebased to resolve conflicts.

**Comment by jrajahalme** [comment]
/test-1.26-net-next

**Comment by jrajahalme** [comment]
Two test fails are for non-required tests, labeling as ready-to-merge.

File: install/kubernetes/cilium/templates/cilium-secrets-namespace.yaml
Comment (asauber) [reply install/kubernetes/cilium/templates/cilium-secrets-namespace.yaml:26]: I much prefer this approach to the current logic, which is not easy to read. This could use a comment along the lines of `{{/* Use a dict to create the set of unique secret namespaces */}}`

**Review by asauber — APPROVED** [review]
Looks good. I'll echo others in saying that the Helm-dict approach to the secrets namespaces would be easier to read and easier to extend.

**Comment by zacharysarah** [comment]
@tklauser I see that this PR merged without the required changes to documentation. Will you please open a followup PR to add the changes that were conditional for approval? Thanks in advance.

**Comment by tklauser** [comment]
The OR was green and ready to merge because CI passed and all reviewers approved. I didn't see any requested changes. Please use the "request changes" review state in case you want to block a PR. Otherwise it's really hard to tell whether changes are still required before merge. Thanks.

**Comment by tklauser** [comment]
PR to address the review comments: https://github.com/cilium/cilium/pull/26296

**Pull request merged**
